internal/service/advancedclustertpf/resource_unit_test.go

```diff
@@ -0,0 +1,239 @@
+
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/service/advancedclustertpf"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/acc"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/testutil/unit"
+)
+
+const (
+	processResponseOnly = "processResponseOnly"
+	projectID           = "111111111111111111111111"
+	clusterName         = "test"
+)
+
+func TestMockAdvancedCluster_replicaset(t *testing.T) {
+	var (
+		oneNewVariable = "backup_enabled = false"
+		fullUpdate     = `
+		backup_enabled = true
+		bi_connector_config = {
+			enabled = true
+		}
+		# config_server_management_mode = "ATLAS_MANAGED" UNSTABLE: After applying this test step, the non-refresh plan was not empty
+		labels = [{
+			key   = "env"
+			value = "test"
+		}]
+		tags = [{
+			key   = "env"
+			value = "test"
+		}]
+		mongo_db_major_version = "8.0"
+		paused = true
+		pit_enabled = true
+		redact_client_log_data = true
+		replica_set_scaling_strategy = "NODE_TYPE"
+		# retain_backups_enabled = true # only set on delete
+		root_cert_type = "ISRGROOTX1"
+		# termination_protection_enabled = true # must be reset to false to enable delete
+		version_release_system = "CONTINUOUS"
+		`
+		// # oplog_min_retention_hours                                      = 5.5
+		// # oplog_size_mb                                                  = 1000
+		// # fail_index_key_too_long 								        = true # only valid for MongoDB version 4.4 and earlier
+		advClusterConfig = `
+		advanced_configuration = {
+			change_stream_options_pre_and_post_images_expire_after_seconds = 100
+			default_read_concern                                           = "available"
+			default_write_concern                                          = "majority"
+			javascript_enabled                                             = false
+			minimum_enabled_tls_protocol                                   = "TLS1_0"
+			no_table_scan                                                  = true
+			sample_refresh_interval_bi_connector                           = 310
+			sample_size_bi_connector                                       = 110
+			transaction_lifetime_limit_seconds                             = 300
+		}
+		`
+		fullUpdateResumed = strings.Replace(fullUpdate, "paused = true", "paused = false", 1)
+		vars              = map[string]string{
+			"groupId":     projectID,
+			"clusterName": clusterName,
+		}
+	)
+	shortenRetries()
+	testCase := resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		Steps: []resource.TestStep{
+			{
+				Config: configBasic(projectID, clusterName, ""),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "state_name", "IDLE"),
+					resource.TestCheckResourceAttr(resourceName, "timeouts.create", "20s"),
+				),
+			},
+			{
+				Config: configBasic(projectID, clusterName, oneNewVariable),
+				Check: resource.ComposeAggregateTestCheckFunc(
```

Comment:
if we only check some properties this is fine.  I wonder if we could check all/most properties in all steps, and then use a helper check func.

Reply:
Adding to https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2840